docs/databases/mysql/use-mysqldump-to-back-up-your-mysql-databases.md

```diff
@@ -0,0 +1,78 @@
+published: 2017-12-06
+title: Use mysqldump to Back Up Your MySQL or MariaDB Databases
+external_resources:
+ - '[MySQL Database Backup Methods page](http://dev.mysql.com/doc/refman/5.1/en/backup-methods.html)'
+ - '[mysqldump - A Database Backup Program, MySQL Reference Manual](https://dev.mysql.com/doc/refman/5.7/en/mysqldump.html)'
+---
+
+![Use mysqldump to Back Up Your MySQL or MariaDB Database](/docs/assets/back_up_your_mysql-databases.png "Use mysqldump to Back Up Your MySQL Database")
+
+[MySQL](http://www.mysql.com/) and [MariaDB](https://mariadb.com/) provide the [mysqldump](https://dev.mysql.com/doc/refman/5.7/en/mysqldump.html) utility to simplify the process of backing up a database or system of databases. Using `mysqldump` is a form of *logical backup*, as opposed to a *physical backup*, which is a copy of the filesystem structure which contains your data. For simplification, the name MySQL will be used throughout this guide when the instructions apply to both MySQL and MariaDB.
+
+
+## Before You Begin
+
+You should have a working installation of MySQL or MariaDB on your system before beginning this guide, and a database user to run the backup as. See our guides for how to install MySQL on Ubuntu or [MariaDB on Debian](/docs/databases/mariadb/mariadb-setup-debian/) for more information.
+
+{{< note >}}
+The steps in this guide require root privileges. Be sure to run the steps below as `root` or with the `sudo` prefix. For more information on privileges, see our [Users and Groups](/docs/tools-reference/linux-users-and-groups) guide.
+{{< /note >}}
+
+## Back up a Database
+
+The backup command's general syntax is shown below. You will be prompted for a password before `mysqldump` begins the backup process. Depending on the size of the database, it could take a while to complete. The database backup will be created in the directory the command is run from, and the `-$(date +%F)` addition to the command will insert a timestamp into the filename.
+
+    mysqldump -u [username] -p [databaseName] > [filename].sql
+
+ - Create a backup of an entire database management system:
+
+        mysqldump --all-databases > full-backup-$(date +%F).sql -u root -p
+
+ - When backing up the entire DBMS isn't necessary, you can dump just a specific database of your choice. Replace `db1` with the name of the database you want to back up.
+
+        mysqldump -u username -p db1 > db1-backup-$(date +%F).sql
+
+- It's also possible to back up a single table from any database. In the example below, `table1` is exported from the database `db1`.
+
+        mysqldump -u username -p db1 table1 > db1-table1-$(date +%F).sql
+
+## Automate Backups with cron
+
+Entries can be added to `/etc/crontab` to regularly schedule database backups. The mysqldump command must be able to run without any user input, so it's important that the syntax of the username and password are given correctly. There should be no space between the `-p` flag and the password.
```

Comment:
@cwlinode, would it be easy enough to put the login elsewhere, so that it's not in the command?

see: [this](https://stackoverflow.com/questions/6861355/mysqldump-launched-by-cron-and-password-security) and the [official docs](https://dev.mysql.com/doc/refman/5.7/en/mysql-config-editor.html) and mysql's [keeping passwords secure](http://ftp.nchu.edu.tw/MySQL/doc/refman/5.0/en/password-security.html)

Reply:
AFAIK, the user and login names need to be in the mysql command in this scenario so cron can execute it.